ui/accordion.info/ui/accordion.reel/accordion.js

```diff
@@ -0,0 +1,152 @@
+ */
+var Montage = require("montage").Montage,
+    Component = require("montage/ui/component").Component;
+
+/**
+ * @class Accordion
+ * @extends Component
+ */
+exports.Accordion = Montage.create(Component, {
+
+    /**
+     * @function prepareForActivationEvents
+     * @param null
+     * @description Add the the event listener for this class object
+     * @private
+     */
+    prepareForActivationEvents: {
+        value: function () {
+            if (window.Touch) {
+                this.element.addEventListener('touchend', this);
+            } else {
+                this.element.addEventListener('mouseup', this);
+            }
+        }
+    },
+
+    /**
+     * @function findClosestOfType
+     * @param el -- Element object
+     * @param type -- Parent flag for element needed find
+     * @param context -- Context for the object
+     * @param clazz -- Flag needed find
+     * @private
+     */
+    _findClosestOfType: {
+        value: function (el, type, context, clazz) {
+            if (el.tagName.toLowerCase() === type) {
+                return el;
+            }
+            var found = false;
+            while (el) {
+                el = el.parentNode;
+                if (el.tagName && el.tagName.toLowerCase() === type) {
+                    if (clazz) {
+                        if (el.classList.contains(clazz)) {
+                            found = true;
+                            break;
+                        }
+                    } else {
+                        found = true;
+                        break;
+                    }
+                }
+                if (!el || !el.ownerDocument || el === context || el.nodeType === 11) {
+                    break;
+                }
+            }
+            if (found) {
+                return el;
+            }
+            return null;
+        }
+    },
+
+    /**
+     *@property headerBackgroundColor
+     *@description Set the background color of header
+     */
+    headerBackgroundColor: {
+        value: null
+    },
+
+    /**
+     * @property title
+     * @description This is the label of header,it is an array.
+     */
+    titles: {
+        value: null
+    },
+
+    /**
+     *@property activeIcon
+     *@description Object of icon for selected status
+     */
+    activeIcon: {
+        value: null
+    },
+
+    /**
+     *@property normalIcon
+     *@description Object of icon for normal status
+     */
+    normalIcon: {
+        value: null
+    },
+
+    /**
+     * @property expendAll
+     * @description Set the property to decide if allow all options are expanded or not, value is boolean.
+     * true, all sections can be expended, false, only one can be expended.
+     */
+    expendAll: {
+        value: false
+    },
+
+    /**
+     * @property selectedOption
+     * @description Set the property to save the selected index of option when expendAll property is false.
+     */
+    selectedOption: {
```

Comment:
Using the repetition, which handle the selection, we get this for free. If the accordion is going to be data driven, then it will have a rangeConroller, created internally from content(an array) or through contentController being set directly. We just had that discussion today with the segmentedBar